packages/store/src/StoreCore.sol

```diff
@@ -476,6 +476,50 @@ library StoreCore {
       return StoreCoreInternal._getDynamicField(tableId, key, schemaIndex, schema);
     }
   }
+
+  /**
+   * Get the byte length of a single field from the given tableId and key tuple, with the given schema
+   */
+  function getFieldLength(
+    bytes32 tableId,
+    bytes32[] memory key,
+    uint8 schemaIndex,
+    Schema schema
+  ) internal view returns (uint256) {
+    uint8 numStaticFields = schema.numStaticFields();
+    if (schemaIndex < numStaticFields) {
+      SchemaType schemaType = schema.atIndex(schemaIndex);
+      return schemaType.getStaticByteLength();
+    } else {
+      // Get the length and storage location of the dynamic field
+      uint8 dynamicSchemaIndex = schemaIndex - numStaticFields;
+      return StoreCoreInternal._loadEncodedDynamicDataLength(tableId, key).atIndex(dynamicSchemaIndex);
+    }
+  }
+
+  /**
+   * Get a byte slice of a single dynamic field from the given tableId and key tuple, with the given schema
+   * The slice is unchecked and will return invalid data if `start`:`end` overflow
+   */
+  function getFieldSlice(
+    bytes32 tableId,
+    bytes32[] memory key,
+    uint8 schemaIndex,
+    Schema schema,
+    uint256 start,
+    uint256 end
```

Comment:
ooc what's the rationale behind start/end vs start/length?

Reply:
mostly consistency: js, solidity (it can only slice calldata though), our `Slice`, and rust all use `start` and `end` (they similarly include start and exclude end, which is why I took it for granted and didn't mention)
imo when working with indexes it's more convenient/clear